flow/BlobCipher.cpp

```diff
@@ -116,74 +116,60 @@ void BlobCipherKey::reset() {
 // BlobKeyIdCache class methods
 
 BlobCipherKeyIdCache::BlobCipherKeyIdCache()
-  : domainId(ENCRYPT_INVALID_DOMAIN_ID), latestBaseCipherKeyId(ENCRYPT_INVALID_CIPHER_KEY_ID),
-    latestRandomSalt(ENCRYPT_INVALID_RANDOM_SALT) {}
+  : domainId(ENCRYPT_INVALID_DOMAIN_ID), latestBaseCipherKeyId(), latestRandomSalt() {}
 
 BlobCipherKeyIdCache::BlobCipherKeyIdCache(EncryptCipherDomainId dId)
-  : domainId(dId), latestBaseCipherKeyId(ENCRYPT_INVALID_CIPHER_KEY_ID), latestRandomSalt(ENCRYPT_INVALID_RANDOM_SALT) {
+  : domainId(dId), latestBaseCipherKeyId(), latestRandomSalt() {
 	TraceEvent("Init_BlobCipherKeyIdCache").detail("DomainId", domainId);
 }
 
 BlobCipherKeyIdCacheKey BlobCipherKeyIdCache::getCacheKey(const EncryptCipherBaseKeyId& baseCipherKeyId,
                                                           const EncryptCipherRandomSalt& salt) {
-	if (baseCipherKeyId == ENCRYPT_INVALID_CIPHER_KEY_ID || salt == ENCRYPT_INVALID_RANDOM_SALT) {
+	if (baseCipherKeyId == ENCRYPT_INVALID_CIPHER_KEY_ID) {
 		throw encrypt_invalid_id();
 	}
 	return std::make_pair(baseCipherKeyId, salt);
 }
 
 Reference<BlobCipherKey> BlobCipherKeyIdCache::getLatestCipherKey() {
-	if (keyIdCache.empty()) {
-		// Cache is empty, nothing more to do.
-		throw encrypt_key_not_found();
+	if (!latestBaseCipherKeyId.present()) {
+		return Reference<BlobCipherKey>();
```

Comment:
it would be good to assert that it is possible iff "keyIdCache.empty()" is true?

Reply:
The cache is not necessary empty. It can be old key(s) was inserted but no latest key in the cache.

Reply:
> The cache is not necessary empty. It can be old key(s) was inserted but no latest key in the cache.

I see, we are talking about the scenario where on bootstrap we have yet only populate cache with point reads; that makes sense and refresh timer has not hit yet. 
Thanks